src/main/scala/io/archivesunleashed/app/ExtractGraphXSLS.scala

```diff
@@ -0,0 +1,73 @@
+/*
+ * Archives Unleashed Toolkit (AUT):
+ * An open-source platform for analyzing web archives.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.archivesunleashed.app
+
+import io.archivesunleashed._
+import io.archivesunleashed.matchbox._
+import org.apache.spark.graphx._
+import org.apache.spark.rdd.RDD
+
+/** Extracts a site link structure using Spark's GraphX utility. */
+object ExtractGraphXSLS {
+
+  /** Creates a hashcode from a url to use as a unique id.
+   *
+   * @param url
+   * @return unique id as long integer.
+   */
+  def pageHash(url: String): VertexId = {
+    url.hashCode.toLong
+  }
```

Comment:
If hash codes collide, the vertex ids won't be unique, right?

Reply:
Yes, that is true. Can you suggest any other way to uniquely assign long values to vertices as an identifier?

Reply:
The hash situation is going to be a problem, especially for very large graphs. 

My research says the solution is some implementation of .zipWithIndex() or .zipWithUniqueId(). I think the latter can cause some problems if the partitions are not the same size, but it does not trigger a spark job. http://spark.apache.org/docs/latest/api/python/pyspark.html#pyspark.RDD.zipWithIndex

I'm not going to try and fix this here however.